Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1429,20 +1443,35 @@ static HRESULT __CGContextRenderToCommandList(CGContextRef context, ID2D1Command
     ComPtr<ID2D1CommandList> commandList;
     RETURN_IF_FAILED(deviceContext->CreateCommandList(&commandList));
 
-    deviceContext->SetTransform(__CGAffineTransformToD2D_F(state.transform));
-
     deviceContext->BeginDraw();
     deviceContext->SetTarget(commandList.Get());
 
+    CGAffineTransform transform = CGAffineTransformIdentity;
+    switch (coordinateMode) {
+        case _kCGCoordinateModeUserSpace:
+            transform = CGContextGetUserSpaceToDeviceSpaceTransform(context);
+            break;
+        case _kCGCoordinateModeDeviceSpace:
+        default:
+            // do nothing; base transform is identity.
+            break;
+    }
+
+    if (additionalTransform) {
+        transform = CGAffineTransformConcat(*additionalTransform, transform);
```

Comment:
The order of the concat matters.  "additional transform" seems to indicate that it should be the second param?